### PR TITLE
Documentation for binding powers and signs

### PR DESCRIPTION
**Pull request by jgosmann**
**Motivation and context:**
This adds the user guide documentation for binding powers and signs (PR #271).

**Interactions with other PRs:**
Based on #271 implementing the actual feature

**How has this been tested?**
Built documentation and looked it.

**How long should this take to review?**

- Average (neither quick nor lengthy)

**Types of changes:**

- Non-code change (touches things like tests, documentation, build scripts)

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [x] I have updated the documentation accordingly.
- [n/a] I have included a changelog entry.
- [n/a] I have added tests to cover my changes.
- [n/a] I have run the test suite locally and all tests passed.

**Still to do:**

* [x] TODOs, documentation for `SemanticPointer` binding powers once implemented.
* [x] proof read

### CODE REVIEW

**Comment by arvoelke** [comment]
One thing I wanted to make sure to post here so that it's not forgotten is this comment https://github.com/nengo/nengo-spa/pull/271#discussion_r595714549 which was the realization that the default implementation for `abs` in the algebra is:
```
return self.bind(self.sign(v).to_vector(len(v)), v)
```
which corresponds perfectly with this definition of sign:
```
sign(z) = z / abs(z)
```
which is the same definition used for integers and complex numbers. That is, if you cross-multiply you get:
```
abs(z) = (1 / sign(z)) * z
```
and then `to_vector` is what is generating `1 / sign(z)`.

File: docs/pyplots/power_similarity.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import nengo_spa as spa
+import matplotlib.pyplot as plt
+
```
Comment (bjkomer) [inline docs/pyplots/power_similarity.py:3]: missing numpy import

**Review by bjkomer — CHANGES_REQUESTED** [review]
LGTM. Just noticed a few typos

**Comment by jgosmann** [comment]
Finally gave this another proof read myself. If this also looks good to you, @bjkomer, @arvoelke, please approve the PR and we can hopefully get the binding powers shipped soon 🚀 .

**Review by arvoelke — APPROVED** [review]

**Pull request merged**